File: 30-days-of-code/Day01/DataTypes.c
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   DataTypes.c                                        :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: alaamimi <alaamimi@student.42.fr>          +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2021/02/04 10:58:40 by alaamimi          #+#    #+#             */
/*   Updated: 2021/02/04 11:02:20 by alaamimi         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <stdio.h>
#include <string.h>
#include <math.h>
#include <stdlib.h>

int main(void)
{
	int i = 4;
	double d = 4.0;
	char s[] = "HackerRank ";

// Declare second integer, double, and String variables.
	int nbr;
	double preci;
	char str[256];
// Read and save an integer, double, and String to your variables.
	scanf("%d", &nbr);
	scanf("%lf", &preci);
	scanf("%*[\n] %[^\n]", &str);
// Print the sum of both integer variables on a new line.
	printf("%d\n", i + nbr);
// Print the sum of the double variables on a new line.
	printf("%.01lf\n", d + preci);
// Concatenate and print the String variables on a new line
	printf("%s%s", s, str);
// The 's' variable above should be printed first.
	return 0;
}

File: 30-days-of-code/Day01/DataTypes.java
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   DataTypes.java                                     :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: alaamimi <alaamimi@student.42.fr>          +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2021/02/24 17:10:27 by alaamimi          #+#    #+#             */
/*   Updated: 2021/02/24 17:10:38 by alaamimi         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

import java.util.Scanner; //Impport Scanner method

public class DataTypes
{
	public static void main(String[] args)
	{
		int i = 4; //Declare first integer
		int integer; //Declare second integer
		double d = 4.0; //declare first double
		double doub; //Declare second double
		String s = "HackerRank "; //Declare first String
		String str; //Declare second String

		Scanner scanInput = new Scanner(System.in); //Declare a Scanner method to take input value from stdin
		//Read and save integer, double && string from stdin
		integer = scanInput.nextInt();
		doub = scanInput.nextDouble();
		scanInput.nextLine(); //Newline character after the double
		str = scanInput.nextLine();

		System.out.println(i + integer); //Print the sum of integer variables on a newline
		System.out.println(d + doub); //Print the sum of doubles on a Newline
		System.out.println(s + str); //concatenate and print the string variables on a newline
		scanInput.close(); //close the input reading
	}
}